test/espnet2/enh/test_espnet_model.py

```diff
@@ -0,0 +1,249 @@
+from distutils.version import LooseVersion
+
+import pytest
+import torch
+
+from espnet2.enh.espnet_model import ESPnetEnhancementModel
+from espnet2.enh.encoder.conv_encoder import ConvEncoder
+from espnet2.enh.encoder.stft_encoder import STFTEncoder
+from espnet2.enh.decoder.conv_decoder import ConvDecoder
+from espnet2.enh.decoder.stft_decoder import STFTDecoder
+from espnet2.enh.separator.dprnn_separator import DPRNNSeparator
+from espnet2.enh.separator.neural_beamformer import NeuralBeamformer
+from espnet2.enh.separator.rnn_separator import RNNSeparator
+from espnet2.enh.separator.tcn_separator import TCNSeparator
+from espnet2.enh.separator.transformer_separator import TransformerSeparator
+
+is_torch_1_2_plus = LooseVersion(torch.__version__) >= LooseVersion("1.2.0")
+
+
+stft_encoder = STFTEncoder(
+    n_fft=28,
+    hop_length=16,
+)
+
+stft_decoder = STFTDecoder(
+    n_fft=28,
+    hop_length=16,
+)
+
+conv_encoder = ConvEncoder(
+    channel=15,
+    kernel_size=32,
+    stride=16,
+)
+
+conv_decoder = ConvDecoder(
+    channel=15,
+    kernel_size=32,
+    stride=16,
+)
+
+rnn_separator = RNNSeparator(
+    input_dim=15,
+    layer=1,
+    unit=10,
+)
+
+dprnn_separator = DPRNNSeparator(input_dim=15, layer=1, unit=10, segment_size=4)
+
+tcn_separator = TCNSeparator(
+    input_dim=15,
+    layer=2,
+    stack=1,
+    bottleneck_dim=10,
+    hidden_dim=10,
+    kernel=3,
+)
+
+transformer_separator = TransformerSeparator(
+    input_dim=15,
+    adim=8,
+    aheads=2,
+    layers=2,
+    linear_units=10,
+)
+
+
+@pytest.mark.parametrize(
+    "encoder, decoder", [(stft_encoder, stft_decoder), (conv_encoder, conv_decoder)]
+)
+@pytest.mark.parametrize(
+    "separator", [rnn_separator, dprnn_separator, tcn_separator, transformer_separator]
+)
+@pytest.mark.parametrize(
+    "loss_type", ["si_snr", "mask_mse", "magnitude", "spectrum", "spectrum_log"]
+)
+@pytest.mark.parametrize("stft_consistency", [True, False])
+@pytest.mark.parametrize("mask_type", ["IBM", "IRM", "IAM", "PSM", "NPSM", "PSM^2"])
+@pytest.mark.parametrize("training", [True, False])
+def test_single_channel_model(
```

Comment:
I have refactored the `ESPnetEnhancementModel` into  `Encoder-Separator-Decoder`.
Now, it supports choosing of `CovEncoder` of `STFTEncdoer`  and any `Separator`.  The usage is shown here in the `ESPnetEnhancementModel` unit test.